intern/cycles/kernel/kernel_passes.h

```diff
@@ -281,46 +282,90 @@ ccl_device_inline void kernel_write_light_passes(KernelGlobals *kg,
     return;
 
   if (light_flag & PASSMASK(DIFFUSE_INDIRECT))
-    kernel_write_pass_float3(buffer + kernel_data.film.pass_diffuse_indirect, L->indirect_diffuse);
+    kernel_write_pass_float3(
```

Comment:
All of these are where we go from the intensities of 3 wavelengths (for the current sample) back to an RGB value so that it can be stored in the output buffer in a compact way. You do lose the spectral information in this process but storing all spectral data for every sample is a massive memory overhead (I know because I did it 😄)

All these passes are just for the different output passes in Cycles. They all need converting back to RGB values, otherwise the intensities for wavelengths 1, 2 and 3 in the current sample will be interpreted as RGB values, and you'll essentially just end up with a grey image.